UM12 - High Gas Use (#265)
NethermindEth/Forta-Agents: Umee-Bots/High-Gas-Use/README.md

# High Amount of Gas Use

## Description

> This bot detects transactions in which high amount of gas is used.

Reviewer: ```suggestion
This bot detects transactions in which high amount of gas is used and specified accounts are involved.
```

The accounts that are needed to be monitored must be added inside the `monitoredAddresses` array in `src/agent.config.ts` file.

The minimum amount of gas that should trigger the alert is set by the `mediumGasThreshold`, `highGasThreshold` and `criticalGasThreshold` values in `src/agent.config.ts` file.

## Supported Chains

- Ethereum

## Alerts

- UMEE-12
  - Fired when a transaction uses high amount of gas
  - Severity is always set to "Medium", "High" or "Critical" depending on the amount of gas use
  - Type is always set to "Suspicious"
  - Metadata:
    - `from`: Address of the initiator of the transaction
    - `to`: Address of the transaction recipient
    - `monitoredAddresses`: Addresses of the monitored accounts that are involved in the transaction
    - `gasUsed`: Amount of gas spent in the transaction

## Test Data

> For testing purposes, `src/agent.config.ts` monitors 'Umee', 'Lending Pool', 'Umee Oracle' and a random user address.

Reviewer: ```suggestion
For testing purposes, the test configurations in `src/agent.config.ts` monitors 'Umee', 'Lending Pool', 'Umee Oracle' and a random user address.
```

### Mainnet

> Uncomment the lines indicated in `src/agent.config.ts` and run:

Reviewer: ```suggestion
Uncomment the lines indicated in `src/agent.config.ts`, add a trace-enabled Mainnet RPC as both `jsonRpcUrl` and `traceRpcUrl` in your `forta.config.json` file and run:
```

```
npm run tx 0x70e41c3279d8014841665f480f1022569ca3fb840a387baa6b69dd8c74f49675
```

This transaction will create a finding in which there are 2 monitored addresses because 2 out of 4 addresses that are monitored are both involved in this transaction.

### Kovan Testnet (PoC)

> In order to check the results, uncomment the lines indicated in `src/agent.config.ts`, set a Kovan testnet RPC (e.g.
> `https://kovan.poa.network`) as `jsonRpcUrl` in your `forta.config.json` file and run:

Reviewer: ```suggestion
In order to check the results, uncomment the lines indicated in `src/agent.config.ts`, set a trace-enabled Kovan testnet RPC as both `jsonRpcUrl` and `traceRpcUrl` in your `forta.config.json` file and run:
```
